SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialReplicationGraph.cpp

```diff
@@ -63,24 +73,63 @@ void USpatialReplicationGraph::OnOwnerUpdated(AActor* Actor, AActor* OldOwner)
 
 void USpatialReplicationGraph::PreReplicateActors(UNetReplicationGraphConnection* ConnectionManager)
 {
+	USpatialNetDriver* SpatialNetDriver = Cast<USpatialNetDriver>(NetDriver);
+	FSpatialReplicationGraphLoadBalancingContext LoadBalancingCtx(SpatialNetDriver, this, ConnectionManager->ActorInfoMap,
+																  PrioritizedReplicationList);
 	if (LoadBalancingHandler.IsValid())
 	{
-		USpatialNetDriver* SpatialNetDriver = Cast<USpatialNetDriver>(NetDriver);
-		FSpatialReplicationGraphLoadBalancingContext LoadBalancingCtx(SpatialNetDriver, this, ConnectionManager->ActorInfoMap,
-																	  PrioritizedReplicationList);
 		LoadBalancingHandler->EvaluateActorsToMigrate(LoadBalancingCtx);
+	}
 
-		for (AActor* Actor : LoadBalancingCtx.AdditionalActorsToReplicate)
+	if (bStrategyWorkerEnabled)
+	{
+		auto& HandoverManager = SpatialNetDriver->HandoverManager;
+		TSet<Worker_EntityId_Key> ActorsToHandover = HandoverManager->GetActorsToHandover();
+		TSet<AActor*> ActorsToConsider;
+		for (Worker_EntityId EntityId : ActorsToHandover)
+		{
+			TWeakObjectPtr<UObject> ObjectPtr = SpatialNetDriver->PackageMap->GetObjectFromEntityId(EntityId);
+			AActor* Actor = Cast<AActor>(ObjectPtr.Get());
+			if (Actor == nullptr)
+			{
+				EntitiesHandedOver.Add(EntityId);
+			}
+			else
+			{
+				if (!ensureAlways(Actor->HasAuthority()) || SpatialNetDriver->LockingPolicy->IsLocked(Actor))
+				{
+					continue;
+				}
+				ActorsToConsider.Add(Actor);
+				ActorsHandedOver.Add(Actor);
+				EntitiesHandedOver.Add(EntityId);
+			}
+		}
+		for (auto ReplicatedActor : LoadBalancingCtx.GetActorsBeingReplicated())
 		{
-			// Only add net owners to the list as they will visit their dependents when replicated.
-			AActor* NetOwner = SpatialGDK::GetReplicatedHierarchyRoot(Actor);
-			if (NetOwner == Actor)
+			AActor& ReplicatedThisFrame = *ReplicatedActor;
+			if (ActorsToConsider.Contains(&ReplicatedThisFrame))
 			{
-				FConnectionReplicationActorInfo& ConnectionData = ConnectionManager->ActorInfoMap.FindOrAdd(Actor);
-				FGlobalActorReplicationInfo& GlobalData = GlobalActorReplicationInfoMap.Get(Actor);
-				PrioritizedReplicationList.Items.Emplace(FPrioritizedRepList::FItem(0, Actor, &GlobalData, &ConnectionData));
+				ActorsToConsider.Remove(&ReplicatedThisFrame);
 			}
 		}
+
+		for (auto ActorToConsider : ActorsToConsider)
+		{
+			LoadBalancingCtx.AddActorToReplicate(ActorToConsider);
+		}
+	}
+
+	for (AActor* Actor : LoadBalancingCtx.AdditionalActorsToReplicate)
```

Comment:
const?

Reply:
Can't be const as it needs to be added to to various containers in non-const form